### PR TITLE
perf: improve performance of AutoMergeBlendShape

### PR DESCRIPTION
**Pull request by bdunderscore**
This pass was obtaining the blendshape buffer for each vertex, then processing those buffers.
However, in practice, most vertices share the same buffer; this pass ignored this, and redid
the same work thousands of times as a result.

This reduces processing time from ~370ms to ~22ms on my main avatar.

### CODE REVIEW

File: Editor/Processors/TraceAndOptimize/AutoMergeBlendShape.cs [hunk 1]
```diff
@@ -43,10 +53,17 @@ private static void DoAutoMerge(MeshInfo2 meshInfo2, BuildContext context)
...
+        {
+            Profiler.EndSample();
+            return;
+        }
+        Profiler.EndSample();
 
+        Profiler.BeginSample("Prepare merge");
         // prepare merge
-        var buffers = meshInfo2.Vertices.Select(x => x.BlendShapeBuffer).ToArray();
+        var buffers = meshInfo2.Vertices.Select(x => x.BlendShapeBuffer).Distinct().ToArray();
```
Comment (bdunderscore) [inline Editor/Processors/TraceAndOptimize/AutoMergeBlendShape.cs:65]: This is the actual change. Should I split it out into a separate commit from all the profiler annotations?

Comment (anatawa12) [reply Editor/Processors/TraceAndOptimize/AutoMergeBlendShape.cs:65]: Yes please

File: CHANGELOG-PRERELEASE.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog].
 ### Added
 
 ### Changed
+- Performance improvements for AutoMergeBlendShape [`#1327`]
```
Comment (anatawa12) [inline CHANGELOG-PRERELEASE.md:13]:
```suggestion
- Performance improvements for AutoMergeBlendShape `#1327`
```

**Review by anatawa12 — APPROVED** [review]
I forgot distinct(). Thank you.

**Comment by anatawa12** [comment]
LGTM. Thank you
